docs/source/services/timelock_service/migration.rst

```diff
@@ -0,0 +1,110 @@
+2. Obtain a timestamp TS that is at least as large as the largest timestamp issued.
+3. Fast-forward the external timestamp service to at least TS.
+4. Invalidate the old timestamp service, preventing AtlasDB clients from retrieving timestamps from it.
+5. Configure AtlasDB clients to retrieve timestamps from the new timestamp service.
+6. Restart AtlasDB clients.
+
+Strictly speaking, step 4 is not required; that said, we prefer to do this to avoid the risk of data corruption in the
+event of misconfiguration (that is, if some but not all clients are pointed to the new timestamp service).
+
+Manual Migration
+----------------
+
+.. danger::
+
+   Ensure that no AtlasDB clients are running during the migration process. Failure to ensure this can result in
+   **SEVERE DATA CORRUPTION** as we are unable to guarantee that timestamps are monotonically increasing!
+
+.. contents::
+   :local:
+
+Step 1: Setting up the Timelock Server
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+First, setup the Timelock Server following the instructions in :ref:`timelock-installation`.
+
+Step 2: Determining the Atlas Timestamp
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+One can use the :ref:`clis` (specifically, the ``timestamp fetch`` command) to obtain a timestamp which is guaranteed
+to be greater than all timestamps issued so far. Make sure that you are requesting for a fresh timestamp as opposed to
+an immutable timestamp (the output should say ``The Fresh timestamp is: N``). Note down the value of this timestamp;
+from here on out, we'll refer to it as N.
+
+Step 3: Fast-Forwarding the Timelock Server
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The Timelock Server exposes an administrative interface, which features a ``fast-forward`` endpoint. Note that this is
+not typically exposed to AtlasDB clients. One can use it to advance the timestamp on the Timelock Server to N, as
+follows:
+
+   .. code:: bash
+
+      curl -XPOST localhost:8080/test/timestamp-admin/fast-forward?newMinimum=N
+
+.. danger::
+
+   Make sure that N has been entered correctly.
+
+    - If N entered is too large, there will generally not be adverse consequences (apart from risks of ``long``
+      overflow), but...
+    - If N entered is too small and AtlasDB clients are restarted without rectifying this, this can result in
+      **SEVERE DATA CORRUPTION** because we lose the guarantee that timestamps are monotonically increasing.
+      As fast-forward is idempotent (it sets the timestamp to be the maximum of its current value and the
+      ``newMinimum``), *if clients have not been started again* this can be fixed by doing another fast-forward.
+
+Step 4: Invalidating the Old Atlas Timestamp
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The steps for invalidating the old Atlas timestamp will vary, depending on your choice of underlying key value store.
+
+- Dropping the table, generally speaking, will *not* work (on the next startup of an embedded Timestamp Service,
```

Comment:
I think it would be better to start with what _does_ work, clearly for each DB. And then dispell the myths of what won't afterwards. I basically think when people read this, they'll look for their KVS, look for your advice, and then perhaps wonder "oh, why can't I just drop the table?", in that order.